Dont rely on the existence of "select" on entry points

rahulporuri opened the pull request:
This PR adds a compatibility layer to the entry point code to handle the absence of the `select` method on entry points. Additionally, we now rely on the external `importlib_metadata` first and only if that is absent do we rely on `importlib.metadata` from the standard library.

fixes #998 

Note to reviewer : I tested this locally by creating a python 3.9 virtual environment, installing pyface and relevent test dependencies and running the testsuite. On this branch, I don't see the `select`-related test errors that we see on the `master` branch.

rahulporuri commented:
@mdickinson do you have the bandwidth to review?

mdickinson commented on pyface/base_toolkit.py:
```diff
@@ -68,11 +68,14 @@
 import sys
 
 try:
+    import importlib_metadata
```
I'd prefer something that defaults to the std. lib. functionality where it's available and only uses `importlib_metadata` as a fallback, since the std. lib. is really what we _want_ to be using: we don't want to be relying on the 3rd party version for any longer than we have to.

With the changes below, we don't actually need to do this reversal, right?

rahulporuri replied:
with these changes, we don't need this reversal but I still chose the reversal specifically we made `importlib_metadata` a required dependency for Python < 3.10 - https://github.com/enthought/pyface/blob/91c93aef0ad034d9060c473457a6ac36514935fd/pyface/__init__.py#L22

mdickinson replied:
Understood, but I'd prefer to get rid of the `importlib_metadata` dependency as soon as we can - i.e., with Python 3.8.

mdickinson reviewed (approved):
LGTM as a quick fix, but I'd rather have a solution that preferred `importlib.metadata` over `importlib_metadata`. That way we can get rid of the 3rd party dependency as soon as we can require Python >= 3.8, and we don't have to wait for Python 3.10.

rahulporuri commented:
@mdickinson can you re-review this PR? I've removed the import reversal - and removed the dependence on external `importlib_metadata` for python >= 3.8

mdickinson reviewed (approved):
LGTM; thanks for the update.

Tested in venvs for Python 3.6, 3.7, 3.8, 3.9 and 3.10(beta), though not surprisingly I wasn't able to get the test suite to run well in 3.10.
